network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -3510,6 +3522,13 @@ private Map<OwnerInfo, List<TapChangerStepAttributes>> getTapChangerStepsWithInC
         if (valuesForInClause.isEmpty()) {
             return Collections.emptyMap();
         }
+        //TODO: to be removed and adapted when tap changer steps are fully migrated — should be after v2.15 deployment
+        Map<OwnerInfo, List<TapChangerStepAttributes>> v214tapChangerSteps = getV214TapChangerStepsWithInClause(this, networkUuid, variantNum, columnNameForWhereClause, valuesForInClause, variantNumOverride);
+        Map<OwnerInfo, List<TapChangerStepAttributes>> newTapChangerSteps = getTapChangerStepsWithInClause(connection, networkUuid, variantNum, columnNameForWhereClause, valuesForInClause, variantNumOverride);
+        return mergeTapChangerSteps(newTapChangerSteps, v214tapChangerSteps);
```

Comment:
Do you really need to merge steps ? If steps are migrated, you get it from the new table. If they are not, you get it from v214 table. For one equipment, we can't have some steps in the old table and some steps in the new table I think